src/connectivity.js

```diff
@@ -137,17 +155,42 @@ var ConnectivityChecker = {
         return reqOptions;
       };
 
+
       if (RdGlobalConfig.proxy) {
+        var proxyOpts = url.parse(RdGlobalConfig.proxy.host + ":" +RdGlobalConfig.proxy.port);
+        if (RdGlobalConfig.proxy.username && RdGlobalConfig.proxy.password) {
+          proxyOpts.auth = RdGlobalConfig.proxy.username + ":" + RdGlobalConfig.proxy.password;
+        }
+
         ConnectivityChecker.connectionChecks.push(this.httpToHubWithProxy, this.httpToRailsWithProxy);
         /* eslint-disable-next-line no-unused-vars */
         ConnectivityChecker.reqOpsWithProxy = function (reqUrl, reqType) {
           var parsedUrl = url.parse(reqUrl);
           var reqOptions = {
             method: 'GET',
             headers: {},
-            host: RdGlobalConfig.proxy.host,
-            port: RdGlobalConfig.proxy.port,
-            path: parsedUrl.href
+            host: parsedUrl.hostname,
+            port: parsedUrl.port || ( reqType === 'http' ? 80 : 443 ),
```

Comment:
no need of the ternary operator now right?...

Reply:
similarly below as well...

Reply:
Done

Reply:
can be resolved.